include/network/Network.h

```diff
@@ -222,27 +261,27 @@ template <typename T> class Network : public NetworkBase<T> {
 
     // Initialize memory with constant values
     static void initialize_constant(DeviceMatrices<T> &ms, const T &constant) { ms.fill(constant); }
+
     void initialize_xavier_normal(DeviceMatrices<T> &ms, const double weight_val_scaling_factor = 1.0) {
         std::random_device rd;
         std::mt19937 gen(rd());
         std::vector<T> weight_matrix_vec(ms.nelements());
         size_t pos = 0;
-        for( uint32_t mat_i = 0; mat_i < ms.GetNumberOfMatrices(); ++mat_i ){
-            int fan_in_out = network_width_+network_width_;
-            if(mat_i == 0){
+        for (uint32_t mat_i = 0; mat_i < ms.GetNumberOfMatrices(); ++mat_i) {
+            int fan_in_out = network_width_ + network_width_;
+            if (mat_i == 0) {
                 // round up to multiple of 16 for compatibility
-                int pad16_width = ((original_input_width_+15)/16)*16;
+                int pad16_width = ((original_input_width_ + 15) / 16) * 16;
                 fan_in_out = pad16_width + network_width_;
-            }
-            else if(mat_i+1 == ms.GetNumberOfMatrices() ){
+            } else if (mat_i + 1 == ms.GetNumberOfMatrices()) {
                 // round up to multiple of 16 for compatibility
-                int pad16_width = ((original_output_width_+15)/16)*16;
+                int pad16_width = ((original_output_width_ + 15) / 16) * 16;
```

Comment:
no hard coded value 16 here. If this 16 is also used anywhere else, make it a constexpr variable of the class. Otherwise take it at least to the beginning of the function.